### ISSUE TITLE
Chado CVTerm service: do we need one or do we just always use Tripal Terms?

### ISSUE THREAD

**Issue by laceysanderson** [tripal/tripal#1780]
### Discussion or Question

This issue is to hold a discussion for whether we want to add a chado cvterm service to support creating, updating, deleting terms in chado. In Tripal 3 these were the following circled cv and db api functions:
![cv api](https://github.com/tripal/tripal/assets/1566301/a223e5e6-8d4f-4fea-8db5-27af7758e617)
![db api](https://github.com/tripal/tripal/assets/1566301/e271544b-ead1-46d6-b911-873b2fc7ad32)

In Tripal 3 we had the above in addition to Tripal Terms which were associated with content types and fields.

In Tripal 4 we already have the Tripal Terms API implemented and it includes a chado storage backend to allow use of chado cvterms directly.

However, we currently use the legacy versions of the functions from the screenshot when creating terms not used for content types and fields --or at least that is what the USASK team is doing in extension modules. These legacy functions are very slow and unreliable so I would like to update them to the new D10 service model...

However, in codefest it came up that maybe we should just use Tripal terms in all cases and we do not need a chado cvterm/db service like we had in Tripal 3.

This issue is to hold discussion on this idea and design if we do go forward with a cvterm/db service.

*Note: I've marked this urgent because the legacy functions provided in Tripal 4 are quite buggy and causing us issues in our extension modules. The urgent part is to decide a course of action :-) I am happy to be the person to implement.*

**Comment by laceysanderson** [comment]
More information:
- Here are the [cv](https://github.com/tripal/tripal/blob/4.x/tripal_chado/src/api/tripal_chado.cv.api.php) and [db](https://github.com/tripal/tripal/blob/4.x/tripal_chado/src/api/tripal_chado.db.api.php) legacy APIs in Tripal 4
- Here is the [Chado Tripal Vocabulary implementation](https://github.com/tripal/tripal/tree/4.x/tripal_chado/src/Plugin/TripalVocabulary) which maps roughly to the chado cv and the [Chado Tripal ID Space implementation](https://github.com/tripal/tripal/blob/4.x/tripal_chado/src/Plugin/TripalIdSpace/ChadoIdSpace.php) which maps roughly to the chado db.
- The Tripal Vocab/ID Space/Term setup is very different from the chado cv/db legacy APIs in that they all work together to handle the whole system of cv, db, dbxref, cvterm whereas the legacy apis where much more piecemeal which allowed you to add just a dbxref without a cvterm, etc.
- If we decide to stick with only the Tripal Vocab/ID Space/Term implementation then we will want to upgrade the legacy api functions to use Tripal Terms rather then contain all the old code.
- If we decide to make a cv/db api service then we will want to use this new service in the Chado implementations of the vocab + idspace classes.

**Comment by laceysanderson** [comment]
Summarizing discussion in codefest:

We've decided to create a `ChadoBuddy` plugin type to provide simple, single table insert, update, upsert, delete functionality for Chado. Then situations like the above are implementations of this plugin type.

I will thus create a ChadoCvtermBuddy and a ChadoDbxrefBuddy plugin implementation which will implement the legacy api functions highlighted above.

To reduce code duplication we will then use these two buddies within the ChadoVocabulary and ChadoIdSpace classes to do the actually querying on chado.

We decided on this route as it will make it easier for people updating from Tripal 3 as they will have a simple mapping from a Tripal 3 function to a Tripal 4 plugin method.

Also, we want to ensure we don't overload existing APIs into large complex structures that are harder to learn and maintain.
